paasta_tools/cli/cmds/generate_pipeline.py

```diff
@@ -43,28 +43,20 @@ def add_subparser(subparsers):
         '-s', '--service',
         help='Name of service for which you wish to generate a Jenkins pipeline',
     ).completer = lazy_choices_completer(list_services)
-    list_parser.add_argument(
-        '-d', '--soa-dir',
-        dest="soa_dir",
-        metavar="SOA_DIR",
-        default=DEFAULT_SOA_DIR,
-        help="define a different soa config directory",
-    )
     list_parser.set_defaults(command=paasta_generate_pipeline)
 
 
 def paasta_generate_pipeline(args):
     """Generate a Jenkins build pipeline.
     :param args: argparse.Namespace obj created from sys.args by cli"""
     service = args.service or guess_service_name()
-    soa_dir = args.soa_dir
     try:
-        validate_service_name(service, soa_dir=soa_dir)
+        validate_service_name(service, soa_dir=DEFAULT_SOA_DIR)
     except NoSuchService as service_not_found:
         print service_not_found
         return 1
 
-    generate_pipeline(service=service, soa_dir=soa_dir)
+    generate_pipeline(service=service)
```

Comment:
Why don't we leave `get_git_repo_for_fab_repo` and `generate_pipeline` (lines 73 and 97) soa-dir aware and just call `generate_pipeline(service=service, soa_dir=DEFAULT_SOA_DIR)`. That way we don't have to change a lot in case we ever decide to make changes to fab_repo.